File: 2021/j2.java
// code by savir singh

import java.util.Scanner;
public class Main {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int N = sc.nextInt();
    int maximum = 0;
    String answer = "";

    for (int i = 0; i < N; i++){
      String name = sc.next();
      int bid = sc.nextInt();

      if (bid > maximum){
        maximum = bid;
        answer = name;
      }
    }

    System.out.println(answer);
    
    
  }
}

File: 2021/j2.cpp
// code by savir singh

#include <iostream>
#include <string>
using namespace std;

int main(){

    int n;
    string winner = "";
    int highest_bid = 0;
    cin >> n;

    for (int i=0; i<n; i++) {
        string name;
        int bid;
        cin >> name;
        cin >> bid;
        if(bid>highest_bid) {
            winner = name;
            highest_bid = bid;
        }
    }
    cout << winner;
    return 0;
}
